File: app/src/main/java/br/com/appdesafio/view/activity/BaseActivity.java
package br.com.appdesafio.view.activity;

import android.os.Bundle;
import android.support.v4.app.Fragment;


import javax.inject.Inject;

import dagger.android.AndroidInjection;
import dagger.android.AndroidInjector;
import dagger.android.DispatchingAndroidInjector;
import dagger.android.support.DaggerAppCompatActivity;

/**
 * activity base, responsible for injecting the activity.
 */
public abstract class BaseActivity extends DaggerAppCompatActivity {

    @Inject
    public DispatchingAndroidInjector<Fragment> dispatchingFragmentInjector;


    @Override
    public void onCreate(final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        AndroidInjection.inject(this);
    }


    @Override
    public AndroidInjector<Fragment> supportFragmentInjector() {
        return dispatchingFragmentInjector;
    }

}

File: app/src/main/java/com/ccortez/desafioconcrete/ui/base/BaseActivity.kt
package com.ccortez.desafioconcrete.ui.base

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import androidx.viewbinding.ViewBinding
import dagger.android.AndroidInjection
import javax.inject.Inject

abstract class BaseActivity<VM : ViewModel, VB : ViewBinding> : AppCompatActivity() {

    @Inject
    protected lateinit var mViewModelProvider: ViewModelProvider.Factory

    protected val mViewModel by lazy {
        getViewModel()
    }

    protected lateinit var mViewBinding: VB


    override fun onCreate(savedInstanceState: Bundle?) {
        AndroidInjection.inject(this)
        super.onCreate(savedInstanceState)

        mViewBinding = getViewBinding()

    }

    abstract fun getViewModel(): VM

    abstract fun getViewBinding(): VB

}